### PR TITLE
[GLUTEN-1299][CH]NativeColumnToRow is used when disable native scan

### PR DESCRIPTION
**Pull request by loudongfeng**
## What changes were proposed in this pull request?

Fixes: \#1299

## How was this patch tested?
Test by adding a UT

### CODE REVIEW

**Comment by kyligence-git** [comment]
Can one of the admins verify this patch?

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala [hunk 1]
```diff
@@ -572,6 +572,19 @@ case class TransformPostOverrides(session: SparkSession, isAdaptiveContext: Bool
       p.withNewChildren(children)
   }
 
+  def nativeSupportedPlan(child: SparkPlan): Boolean = {
```
Comment (jinchengchenghh) [inline gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:575]: Can you move this logic to `nativeConversion.doValidate()`

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala [hunk 2]
```diff
@@ -572,6 +572,19 @@ case class TransformPostOverrides(session: SparkSession, isAdaptiveContext: Bool
...
+    if (child.isInstanceOf[TransformSupport]) {
```
Comment (PHILO-HE) [inline gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:576]: There are some native operators that not extending `TransformSupport`, e.g., `ColumnarShuffleExchangeExec`. Recently, we have a common trait added, named `GlutenPlan`. Maybe, we should use it for checking. Thanks!

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala [hunk 3]
```diff
@@ -572,6 +572,19 @@ case class TransformPostOverrides(session: SparkSession, isAdaptiveContext: Bool
...
+      return true
+    }
+    if (!child.supportsColumnar) {
```
Comment (PHILO-HE) [inline gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:579]: As we know, `ColumnarToRow` is inserted by spark with child plan checked like this line. So I think we can remove the redundant check here. Right?

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala (hunk 1 above)
Comment (PHILO-HE) [reply gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:575]: +1

File: backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseParquetSuite.scala [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  override protected def sparkConf: SparkConf = {
+    super.sparkConf
+      .set(GlutenConfig.GLUTEN_LIB_PATH, UTSystemParameters.getClickHouseLibPath())
+      .set("spark.gluten.sql.enable.native.validation", "false")
+      .set("spark.sql.warehouse.dir", warehouse)
+  }
+
+  test("native scan disabled case") {
+    withSQLConf(("spark.gluten.enabled", "false")) {
```
Comment (zzcclp) [inline backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseParquetSuite.scala:43]: why is disable gluten ?

Comment (loudongfeng) [reply backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseParquetSuite.scala:43]: Use sparksql to create data first, then run actual test bellow.Just to make sure no error will occur.

**Comment by loudongfeng** [comment]
It is inconvenient to run velox backend's test cases in my local environment for now.So just trigger CI to see what will happen.

File: gluten-data/src/main/scala/org/apache/spark/sql/execution/GlutenColumnarToRowExec.scala [hunk 5]
```diff
@@ -46,6 +48,16 @@ case class GlutenColumnarToRowExec(child: SparkPlan)
   override def nodeName: String = "GlutenColumnarToRowExec"
 
   override def buildCheck(): Unit = {
+    if (
+      !child.isInstanceOf[GlutenPlan]
+      // columnar shuffle may use spark's sort shuffle manager
```
[the remark this answers is not in the document]
Comment (loudongfeng) [reply gluten-data/src/main/scala/org/apache/spark/sql/execution/GlutenColumnarToRowExec.scala:53]: Thanks @PHILO-HE . I only find this issue for gluten+ch+parquet, will only keep the check for ch backend.

**Comment by zhztheplayer** [comment]
Is the patch now only for CH backend?

**Comment by zzcclp** [comment]
please rebase, and review later again.

**Pull request closed without merging**